File: archive/postman-march-2026/postman/sdks/Collections SDKs/tagslut - API/java/src/main/java/com/tagslutapisdk/services/IdentityVerificationService.java
package com.tagslutapisdk.services;

import com.fasterxml.jackson.core.type.TypeReference;
import com.tagslutapisdk.config.RequestConfig;
import com.tagslutapisdk.config.TagslutApiSdkConfig;
import com.tagslutapisdk.exceptions.ApiError;
import com.tagslutapisdk.http.Environment;
import com.tagslutapisdk.http.HttpMethod;
import com.tagslutapisdk.http.ModelConverter;
import com.tagslutapisdk.http.util.RequestBuilder;
import com.tagslutapisdk.models._5aBeatportIsrcLookupParameters;
import com.tagslutapisdk.models._5bTidalIsrcCrossCheckParameters;
import com.tagslutapisdk.models._5cSpotifyIsrcCrossCheckParameters;
import java.util.Optional;
import java.util.concurrent.CompletableFuture;
import lombok.NonNull;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

/**
 * IdentityVerificationService Service
 */
public class IdentityVerificationService extends BaseService {

  private RequestConfig _5aBeatportIsrcLookupConfig = RequestConfig.builder()
    .environment(Environment.BASE_URL_1)
    .build();
  private RequestConfig _5bTidalIsrcCrossCheckConfig = RequestConfig.builder()
    .environment(Environment.API)
    .build();
  private RequestConfig _5cSpotifyIsrcCrossCheckConfig = RequestConfig.builder()
    .environment(Environment.API_1)
    .build();

  /**
   * Constructs a new instance of IdentityVerificationService.
   *
   * @param httpClient The HTTP client to use for requests
   * @param config The SDK configuration
   */
  public IdentityVerificationService(@NonNull OkHttpClient httpClient, TagslutApiSdkConfig config) {
    super(httpClient, config);
  }

  /**
   * Sets method-level configuration for {@code _5aBeatportIsrcLookup}.
   * Method-level overrides take precedence over service-level configuration but are
   * overridden by request-level configurations.
   *
   * @param config The configuration overrides to apply at the method level
   * @return This service instance for method chaining
   */
  public IdentityVerificationService set_5aBeatportIsrcLookupConfig(RequestConfig config) {
    this._5aBeatportIsrcLookupConfig = config;
    return this;
  }

  /**
   * Sets method-level configuration for {@code _5bTidalIsrcCrossCheck}.
   * Method-level overrides take precedence over service-level configuration but are
   * overridden by request-level configurations.
   *
   * @param config The configuration overrides to apply at the method level
   * @return This service instance for method chaining
   */
  public IdentityVerificationService set_5bTidalIsrcCrossCheckConfig(RequestConfig config) {
    this._5bTidalIsrcCrossCheckConfig = config;
    return this;
  }

  /**
   * Sets method-level configuration for {@code _5cSpotifyIsrcCrossCheck}.
   * Method-level overrides take precedence over service-level configuration but are
   * overridden by request-level configurations.
   *
   * @param config The configuration overrides to apply at the method level
   * @return This service instance for method chaining
   */
  public IdentityVerificationService set_5cSpotifyIsrcCrossCheckConfig(RequestConfig config) {
    this._5cSpotifyIsrcCrossCheckConfig = config;
    return this;
  }

  /**
   * Method _5aBeatportIsrcLookup
   * GET /v4/catalog/tracks
   *
   * @return response of {@code Object}
   */
  public Object _5aBeatportIsrcLookup() throws ApiError {
    return this._5aBeatportIsrcLookup(_5aBeatportIsrcLookupParameters.builder().build());
  }

  /**
   * Method _5aBeatportIsrcLookup
   * GET /v4/catalog/tracks
   *
   * @param requestParameters {@link _5aBeatportIsrcLookupParameters} Request Parameters Object
   * @return response of {@code Object}
   */
  public Object _5aBeatportIsrcLookup(@NonNull _5aBeatportIsrcLookupParameters requestParameters)
    throws ApiError {
    return this._5aBeatportIsrcLookup(requestParameters, null);
  }

  /**
   * Method _5aBeatportIsrcLookup
   * GET /v4/catalog/tracks
   *
   * @param requestParameters {@link _5aBeatportIsrcLookupParameters} Request Parameters Object
   * @return response of {@code Object}
   */
  public Object _5aBeatportIsrcLookup(
    @NonNull _5aBeatportIsrcLookupParameters requestParameters,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._5aBeatportIsrcLookupConfig, requestConfig);
    Request request = this.build_5aBeatportIsrcLookupRequest(requestParameters, resolvedConfig);
    Response response = this.execute(request, resolvedConfig);
    byte[] bodyBytes = ModelConverter.readBytes(response);
    return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
  }

  /**
   * Method _5aBeatportIsrcLookup
   * GET /v4/catalog/tracks
   *
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5aBeatportIsrcLookupAsync() throws ApiError {
    return this._5aBeatportIsrcLookupAsync(_5aBeatportIsrcLookupParameters.builder().build());
  }

  /**
   * Method _5aBeatportIsrcLookup
   * GET /v4/catalog/tracks
   *
   * @param requestParameters {@link _5aBeatportIsrcLookupParameters} Request Parameters Object
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5aBeatportIsrcLookupAsync(
    @NonNull _5aBeatportIsrcLookupParameters requestParameters
  ) throws ApiError {
    return this._5aBeatportIsrcLookupAsync(requestParameters, null);
  }

  /**
   * Method _5aBeatportIsrcLookup
   * GET /v4/catalog/tracks
   *
   * @param requestParameters {@link _5aBeatportIsrcLookupParameters} Request Parameters Object
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5aBeatportIsrcLookupAsync(
    @NonNull _5aBeatportIsrcLookupParameters requestParameters,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._5aBeatportIsrcLookupConfig, requestConfig);
    Request request = this.build_5aBeatportIsrcLookupRequest(requestParameters, resolvedConfig);
    CompletableFuture<Response> futureResponse = this.executeAsync(request, resolvedConfig);
    return futureResponse.thenApplyAsync(response -> {
      byte[] bodyBytes = ModelConverter.readBytes(response);
      return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
    });
  }

  private Request build_5aBeatportIsrcLookupRequest(
    @NonNull _5aBeatportIsrcLookupParameters requestParameters,
    RequestConfig resolvedConfig
  ) {
    return new RequestBuilder(
      HttpMethod.GET,
      resolveBaseUrl(resolvedConfig, Environment.BASE_URL_1),
      "v4/catalog/tracks"
    )
      .setAccessTokenAuth(resolveAccessToken(resolvedConfig))
      .setOptionalQueryParameter("isrc", requestParameters.getIsrc())
      .build();
  }

  /**
   * Method _5bTidalIsrcCrossCheck
   * GET /v1/tracks
   *
   * @return response of {@code Object}
   */
  public Object _5bTidalIsrcCrossCheck() throws ApiError {
    return this._5bTidalIsrcCrossCheck(_5bTidalIsrcCrossCheckParameters.builder().build());
  }

  /**
   * Method _5bTidalIsrcCrossCheck
   * GET /v1/tracks
   *
   * @param requestParameters {@link _5bTidalIsrcCrossCheckParameters} Request Parameters Object
   * @return response of {@code Object}
   */
  public Object _5bTidalIsrcCrossCheck(@NonNull _5bTidalIsrcCrossCheckParameters requestParameters)
    throws ApiError {
    return this._5bTidalIsrcCrossCheck(requestParameters, null);
  }

  /**
   * Method _5bTidalIsrcCrossCheck
   * GET /v1/tracks
   *
   * @param requestParameters {@link _5bTidalIsrcCrossCheckParameters} Request Parameters Object
   * @return response of {@code Object}
   */
  public Object _5bTidalIsrcCrossCheck(
    @NonNull _5bTidalIsrcCrossCheckParameters requestParameters,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._5bTidalIsrcCrossCheckConfig, requestConfig);
    Request request = this.build_5bTidalIsrcCrossCheckRequest(requestParameters, resolvedConfig);
    Response response = this.execute(request, resolvedConfig);
    byte[] bodyBytes = ModelConverter.readBytes(response);
    return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
  }

  /**
   * Method _5bTidalIsrcCrossCheck
   * GET /v1/tracks
   *
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5bTidalIsrcCrossCheckAsync() throws ApiError {
    return this._5bTidalIsrcCrossCheckAsync(_5bTidalIsrcCrossCheckParameters.builder().build());
  }

  /**
   * Method _5bTidalIsrcCrossCheck
   * GET /v1/tracks
   *
   * @param requestParameters {@link _5bTidalIsrcCrossCheckParameters} Request Parameters Object
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5bTidalIsrcCrossCheckAsync(
    @NonNull _5bTidalIsrcCrossCheckParameters requestParameters
  ) throws ApiError {
    return this._5bTidalIsrcCrossCheckAsync(requestParameters, null);
  }

  /**
   * Method _5bTidalIsrcCrossCheck
   * GET /v1/tracks
   *
   * @param requestParameters {@link _5bTidalIsrcCrossCheckParameters} Request Parameters Object
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5bTidalIsrcCrossCheckAsync(
    @NonNull _5bTidalIsrcCrossCheckParameters requestParameters,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._5bTidalIsrcCrossCheckConfig, requestConfig);
    Request request = this.build_5bTidalIsrcCrossCheckRequest(requestParameters, resolvedConfig);
    CompletableFuture<Response> futureResponse = this.executeAsync(request, resolvedConfig);
    return futureResponse.thenApplyAsync(response -> {
      byte[] bodyBytes = ModelConverter.readBytes(response);
      return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
    });
  }

  private Request build_5bTidalIsrcCrossCheckRequest(
    @NonNull _5bTidalIsrcCrossCheckParameters requestParameters,
    RequestConfig resolvedConfig
  ) {
    return new RequestBuilder(
      HttpMethod.GET,
      resolveBaseUrl(resolvedConfig, Environment.API),
      "v1/tracks"
    )
      .setAccessTokenAuth(resolveAccessToken(resolvedConfig))
      .setOptionalQueryParameter("isrc", requestParameters.getIsrc())
      .setOptionalQueryParameter("countryCode", requestParameters.getCountryCode())
      .build();
  }

  /**
   * Method _5cSpotifyIsrcCrossCheck
   * GET /v1/search
   *
   * @return response of {@code Object}
   */
  public Object _5cSpotifyIsrcCrossCheck() throws ApiError {
    return this._5cSpotifyIsrcCrossCheck(_5cSpotifyIsrcCrossCheckParameters.builder().build());
  }

  /**
   * Method _5cSpotifyIsrcCrossCheck
   * GET /v1/search
   *
   * @param requestParameters {@link _5cSpotifyIsrcCrossCheckParameters} Request Parameters Object
   * @return response of {@code Object}
   */
  public Object _5cSpotifyIsrcCrossCheck(
    @NonNull _5cSpotifyIsrcCrossCheckParameters requestParameters
  ) throws ApiError {
    return this._5cSpotifyIsrcCrossCheck(requestParameters, null);
  }

  /**
   * Method _5cSpotifyIsrcCrossCheck
   * GET /v1/search
   *
   * @param requestParameters {@link _5cSpotifyIsrcCrossCheckParameters} Request Parameters Object
   * @return response of {@code Object}
   */
  public Object _5cSpotifyIsrcCrossCheck(
    @NonNull _5cSpotifyIsrcCrossCheckParameters requestParameters,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._5cSpotifyIsrcCrossCheckConfig, requestConfig);
    Request request = this.build_5cSpotifyIsrcCrossCheckRequest(requestParameters, resolvedConfig);
    Response response = this.execute(request, resolvedConfig);
    byte[] bodyBytes = ModelConverter.readBytes(response);
    return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
  }

  /**
   * Method _5cSpotifyIsrcCrossCheck
   * GET /v1/search
   *
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5cSpotifyIsrcCrossCheckAsync() throws ApiError {
    return this._5cSpotifyIsrcCrossCheckAsync(_5cSpotifyIsrcCrossCheckParameters.builder().build());
  }

  /**
   * Method _5cSpotifyIsrcCrossCheck
   * GET /v1/search
   *
   * @param requestParameters {@link _5cSpotifyIsrcCrossCheckParameters} Request Parameters Object
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5cSpotifyIsrcCrossCheckAsync(
    @NonNull _5cSpotifyIsrcCrossCheckParameters requestParameters
  ) throws ApiError {
    return this._5cSpotifyIsrcCrossCheckAsync(requestParameters, null);
  }

  /**
   * Method _5cSpotifyIsrcCrossCheck
   * GET /v1/search
   *
   * @param requestParameters {@link _5cSpotifyIsrcCrossCheckParameters} Request Parameters Object
   * @return response of {@code CompletableFuture<Object>}
   */
  public CompletableFuture<Object> _5cSpotifyIsrcCrossCheckAsync(
    @NonNull _5cSpotifyIsrcCrossCheckParameters requestParameters,
    RequestConfig requestConfig
  ) throws ApiError {
    RequestConfig resolvedConfig =
      this.getResolvedConfig(this._5cSpotifyIsrcCrossCheckConfig, requestConfig);
    Request request = this.build_5cSpotifyIsrcCrossCheckRequest(requestParameters, resolvedConfig);
    CompletableFuture<Response> futureResponse = this.executeAsync(request, resolvedConfig);
    return futureResponse.thenApplyAsync(response -> {
      byte[] bodyBytes = ModelConverter.readBytes(response);
      return ModelConverter.convert(bodyBytes, new TypeReference<Object>() {});
    });
  }

  private Request build_5cSpotifyIsrcCrossCheckRequest(
    @NonNull _5cSpotifyIsrcCrossCheckParameters requestParameters,
    RequestConfig resolvedConfig
  ) {
    return new RequestBuilder(
      HttpMethod.GET,
      resolveBaseUrl(resolvedConfig, Environment.API_1),
      "v1/search"
    )
      .setAccessTokenAuth(resolveAccessToken(resolvedConfig))
      .setOptionalQueryParameter("q", requestParameters.getQ())
      .setOptionalQueryParameter("type", requestParameters.getType())
      .build();
  }
}

File: postman/sdks/Collections SDKs/tagslut - API/typescript/src/services/identity-verification/identity-verification-service.ts
import { z } from 'zod';
import { BaseService } from '../base-service';
import { ContentType, HttpResponse, SdkConfig } from '../../http/types';
import { RequestBuilder } from '../../http/transport/request-builder';
import { SerializationStyle } from '../../http/serialization/base-serializer';
import { ThrowableError } from '../../http/errors/throwable-error';
import { Environment } from '../../http/environment';
import {
  _5aBeatportIsrcLookupParams,
  _5bTidalIsrcCrossCheckParams,
  _5cSpotifyIsrcCrossCheckParams,
} from './request-params';

/**
 * Service class for IdentityVerificationService operations.
 * Provides methods to interact with IdentityVerificationService-related API endpoints.
 * All methods return promises and handle request/response serialization automatically.
 */
export class IdentityVerificationService extends BaseService {
  protected _5aBeatportIsrcLookupConfig: Partial<SdkConfig> = {
    environment: Environment.BASE_URL_1,
  };

  protected _5bTidalIsrcCrossCheckConfig: Partial<SdkConfig> = { environment: Environment.API };

  protected _5cSpotifyIsrcCrossCheckConfig: Partial<SdkConfig> = { environment: Environment.API_1 };

  /**
   * Sets method-level configuration for _5aBeatportIsrcLookup.
   * @param config - Partial configuration to override service-level defaults
   * @returns This service instance for method chaining
   */
  set_5aBeatportIsrcLookupConfig(config: Partial<SdkConfig>): this {
    this._5aBeatportIsrcLookupConfig = config;
    return this;
  }

  /**
   * Sets method-level configuration for _5bTidalIsrcCrossCheck.
   * @param config - Partial configuration to override service-level defaults
   * @returns This service instance for method chaining
   */
  set_5bTidalIsrcCrossCheckConfig(config: Partial<SdkConfig>): this {
    this._5bTidalIsrcCrossCheckConfig = config;
    return this;
  }

  /**
   * Sets method-level configuration for _5cSpotifyIsrcCrossCheck.
   * @param config - Partial configuration to override service-level defaults
   * @returns This service instance for method chaining
   */
  set_5cSpotifyIsrcCrossCheckConfig(config: Partial<SdkConfig>): this {
    this._5cSpotifyIsrcCrossCheckConfig = config;
    return this;
  }

  /**
   *
   * @param {string} [params.isrc] -
   * @param {Partial<SdkConfig>} [requestConfig] - The request configuration for retry and validation.
   * @returns {Promise<HttpResponse<any>>} - OK
   */
  async _5aBeatportIsrcLookup(
    params?: _5aBeatportIsrcLookupParams,
    requestConfig?: Partial<SdkConfig>,
  ): Promise<any> {
    const resolvedConfig = this.getResolvedConfig(this._5aBeatportIsrcLookupConfig, requestConfig);
    const request = new RequestBuilder()
      .setConfig(resolvedConfig)
      .setBaseUrl(resolvedConfig)
      .setMethod('GET')
      .setPath('/v4/catalog/tracks')
      .setRequestSchema(z.any())
      .addAccessTokenAuth(resolvedConfig?.token, 'Bearer')
      .setRequestContentType(ContentType.Json)
      .addResponse({
        schema: z.any(),
        contentType: ContentType.Json,
        status: 200,
      })
      .addQueryParam({
        key: 'isrc',
        value: params?.isrc,
      })
      .build();
    return this.client.callDirect<any>(request);
  }

  /**
   *
   * @param {string} [params.isrc] -
   * @param {string} [params.countryCode] -
   * @param {Partial<SdkConfig>} [requestConfig] - The request configuration for retry and validation.
   * @returns {Promise<HttpResponse<any>>} - OK
   */
  async _5bTidalIsrcCrossCheck(
    params?: _5bTidalIsrcCrossCheckParams,
    requestConfig?: Partial<SdkConfig>,
  ): Promise<any> {
    const resolvedConfig = this.getResolvedConfig(this._5bTidalIsrcCrossCheckConfig, requestConfig);
    const request = new RequestBuilder()
      .setConfig(resolvedConfig)
      .setBaseUrl(resolvedConfig)
      .setMethod('GET')
      .setPath('/v1/tracks')
      .setRequestSchema(z.any())
      .addAccessTokenAuth(resolvedConfig?.token, 'Bearer')
      .setRequestContentType(ContentType.Json)
      .addResponse({
        schema: z.any(),
        contentType: ContentType.Json,
        status: 200,
      })
      .addQueryParam({
        key: 'isrc',
        value: params?.isrc,
      })
      .addQueryParam({
        key: 'countryCode',
        value: params?.countryCode,
      })
      .build();
    return this.client.callDirect<any>(request);
  }

  /**
   *
   * @param {string} [params.q] -
   * @param {string} [params.type] -
   * @param {Partial<SdkConfig>} [requestConfig] - The request configuration for retry and validation.
   * @returns {Promise<HttpResponse<any>>} - OK
   */
  async _5cSpotifyIsrcCrossCheck(
    params?: _5cSpotifyIsrcCrossCheckParams,
    requestConfig?: Partial<SdkConfig>,
  ): Promise<any> {
    const resolvedConfig = this.getResolvedConfig(
      this._5cSpotifyIsrcCrossCheckConfig,
      requestConfig,
    );
    const request = new RequestBuilder()
      .setConfig(resolvedConfig)
      .setBaseUrl(resolvedConfig)
      .setMethod('GET')
      .setPath('/v1/search')
      .setRequestSchema(z.any())
      .addAccessTokenAuth(resolvedConfig?.token, 'Bearer')
      .setRequestContentType(ContentType.Json)
      .addResponse({
        schema: z.any(),
        contentType: ContentType.Json,
        status: 200,
      })
      .addQueryParam({
        key: 'q',
        value: params?.q,
      })
      .addQueryParam({
        key: 'type',
        value: params?.type,
      })
      .build();
    return this.client.callDirect<any>(request);
  }
}
